### ISSUE TITLE
torch.ldexp drops the derivative contribution of a negative exponent

### ISSUE THREAD

**Issue by ALinrunrun** [pytorch/pytorch#196680]
### 🐛 Describe the bug

`torch.ldexp` returns `2.0` for a first-order derivative whose mathematically correct value is `2.35` in `float64`.

The integer exponents are fixed; no derivative with respect to an integer is requested. For inputs `[t, t**2]` and exponents `[1, -2]`, `ldexp` computes `2*t + t**2/4`, so the first derivative is `2 + t/2`. At `t = 0.7`, that derivative is `2.35`. PyTorch returns `2.0`, which is only the contribution of the positive exponent.

This issue reproduces on CPU, so CUDA availability does not affect the result.

```python
import torch

dtype = torch.float64

t = torch.tensor(0.7, dtype=dtype, requires_grad=True)
x = torch.stack((t, t * t))
e = torch.tensor([1, -2], dtype=torch.int32, device=t.device)

y = torch.ldexp(x, e).sum()
actual, = torch.autograd.grad(y, t)

expected = torch.tensor(2.35, dtype=dtype)

print("actual:")
print(actual)
print("expected:")
print(expected)
```

### Actual vs expected result

```text
actual:
2.0

expected:
2.35
```

The discrepancy occurs in the first-order backward pass. The forward value is `1.5225` and matches `2*t + t**2/4`, but the derivative of the negative-exponent term `t**2/4` is dropped, so the gradient is `2.0` instead of `2.35`.

### Versions

```text
PyTorch version: 2.12.0+cu130
Is debug build: False
CUDA used to build PyTorch: 13.0
ROCM used to build PyTorch: N/A

OS: Ubuntu 24.04.3 LTS (x86_64)
Python version: 3.13.13 | packaged by Anaconda, Inc.
Python platform: Linux x86_64

Is CUDA available: False
GPU models:
  GPU 0: NVIDIA RTX 6000 Ada Generation
  GPU 1: NVIDIA RTX 6000 Ada Generation
Nvidia driver version: 570.211.01

CPU:
  Model name: AMD Ryzen Threadripper PRO 7985WX 64-Cores
  CPU(s): 128

Versions of relevant libraries:
[pip3] numpy==2.4.6
[pip3] torch==2.12.0
[pip3] triton==3.7.0
```

cc @ezyang @albanD @gqchen @nikitaved @soulitzer @Varal7 @bobrenjc93

**Comment by ngimel** [comment]
Doesn't repro on master
